[excel] (Tutorial) Adding links to the completed version of the tutorial (#3801)
OfficeDev/office-js-docs-pr: docs/tutorials/excel-tutorial.md

> [!TIP]
> If you've already completed the [Build an Excel task pane add-in](../quickstarts/excel-quickstart-jquery.md) quick start using the Yeoman generator, and want to use that project as a starting point for this tutorial, go directly to the [Create a table](#create-a-table) section to start this tutorial.
>
> If you want a completed version of this tutorial, head over to the [Office Add-ins samples repo on GitHub](https://github.com/OfficeDev/Office-Add-in-samples/tree/main/Samples/tutorials/excel-tutorial).

## Prerequisites

> 

Reviewer: Consider including the instruction to run "npm install", perhaps at the end of the "Create your add-in project" section. Without running that instruction, I ended hitting the following error when I tried to test the add-in later on:
```
'office-addin-debugging' is not recognized as an internal or external command,
operable program or batch file.
```

1. Within the `Office.onReady` function call, locate the line `if (info.host === Office.HostType.Excel) {` and add the following code immediately after that line. Note:

    - This code adds an event handler for the `create-table` button.
    - The `createTable` function is wrapped in a call to `tryCatch` (both functions will be added next step). This allows any errors generated by the Office JavaScript layer to be handled separate from your service code.

    ```js
    // Assign event handlers and other initialization logic.
    document.getElementById("create-table").onclick = () => tryCatch(createTable);
    ```

1. Add the following functions to the end of the file. Note:

    - Your Excel.js business logic will be added to the function that is passed to `Excel.run`. This logic does not execute immediately. Instead, it is added to a queue of pending commands.

    - The `context.sync` method sends all queued commands to Excel for execution.

>     - The `tryCatch` function will be used by all the functions interacting with the workbook from the taskpane. Catching Office JavaScript errors in this fashion is a convenient way to generically handle any uncaught errors.

Reviewer: ```suggestion
    - The `tryCatch` function will be used by all the functions interacting with the workbook from the task pane. Catching Office JavaScript errors in this fashion is a convenient way to generically handle any uncaught errors.
```

            // TODO3: Queue commands to format the table.

            await context.sync();
        });
    }

> 

Author: I'll downgrade this to a note. We added this comment because of customer feedback when we initially changed the JS to ES6.

> 1. Within the `Office.onReady` function call, locate the line that assigns a click handler to the `sort-table` button, and add the following code after that line.

Author: That is intended to say `AnyCollection`. It's similar to how we denote things elsewhere: https://learn.microsoft.com/en-us/office/dev/add-ins/develop/application-specific-api-model#run-function

    ```js
    async function toggleProtection(args) {
        try {
            await Excel.run(async (context) => {

                // TODO1: Queue commands to reverse the protection status of the current worksheet.

                await context.sync();
            });
        } catch (error) {
            // Note: In a production add-in, you'd want to notify the user through your add-in's UI.
            console.error(error);
        }

        args.completed();
    }
    ```

> 1. Add the following line immediately below the function to register it.

Reviewer: ```suggestion
1. Add the following line immediately after the function to register it.
```

    ```js
    async function toggleProtection(args) {
        try {
            await Excel.run(async (context) => {
                const sheet = context.workbook.worksheets.getActiveWorksheet();

                sheet.load('protection/protected');
                await context.sync();
            
                if (sheet.protection.protected) {
                    sheet.protection.unprotect();
                } else {
                    sheet.protection.protect();
                }

                await context.sync();
            });
        } catch (error) {
            // Note: In a production add-in, you'd want to notify the user through your add-in's UI.
            console.error(error);
        }

        args.completed();
    }

>     ```

Reviewer: For the [Test the add-in](https://review.learn.microsoft.com/en-us/office/dev/add-ins/tutorials/excel-tutorial?branch=pr-en-us-3801#test-the-add-in-4) section for "Protect a worksheet", the first 2 steps apply for desktop but not for web. We should include the web instructions (could just be linking to the [appropriate section](https://learn.microsoft.com/en-us/office/dev/add-ins/testing/sideload-office-add-ins-for-testing?branch=pr-en-us-1491#remove-a-sideloaded-add-in) of the "Sideload..." article.

Also, step 4 of that section should be more similar to the other "Test the add-in" sections, particularly for the web instructions

For step 6, is the behavior different between desktop and web? In the web version, you have to try changing a cell (e.g., typing in some text) for the msg to pop up. I haven't tried desktop yet

1. Add the following code to **popup.js**. Note the following about this code.

   - *Every page that calls APIs in the Office.js library must first ensure that the library is fully initialized.* The best way to do that is to call the `Office.onReady()` function. The call of `Office.onReady()` must run before any calls to Office.js; hence the assignment is in a script file that is loaded by the page, as it is in this case.

    ```js
    Office.onReady((info) => {
        // TODO1: Assign handler to the OK button.
    });

>     // TODO2: Create the OK button handler

Reviewer: ```suggestion
    // TODO2: Create the OK button handler.
```

> - [Office Add-ins platform overview](../overview/office-add-ins.md)

Reviewer: nit: In step 7 of the [Open the dialog from the task pane](https://review.learn.microsoft.com/en-us/office/dev/add-ins/tutorials/excel-tutorial?branch=pr-en-us-3801#open-the-dialog-from-the-task-pane) section, the TODO1 comment should end with a period.

> - [Office Add-ins platform overview](../overview/office-add-ins.md)

Reviewer: For step 6 of the final [Test the add-in](https://review.learn.microsoft.com/en-us/office/dev/add-ins/tutorials/excel-tutorial?branch=pr-en-us-3801#test-the-add-in-5) section, it would be useful to indicate which file contains the `processMessage` function.

Author: Did you have a different file open? Reading through, the file with this method seems to be the last one edited.

Reviewer: I think I may have had a different file open because I clicked around checking on things then got to that step and wasn't sure which of the many tabs was the file I needed to update
